src/App.tsx

```diff
@@ -1,9 +1,55 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import classNames from 'classnames';
+import React, { useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import { USER_ID, getTodos } from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { wait } from './utils/fetchClient';
+import { Status } from './types/Status';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [search, setSearch] = useState<string>('');
+  const [sortBy, setSortBy] = useState<Status>(Status.All);
+
+  const titleInput = useRef<HTMLInputElement>(null);
+
+  const isAllCompleted = todos.every(todo => todo.completed);
+
+  const activeTodosCount = todos.filter(todo => !todo.completed).length;
+
+  let visibleTodos = todos.filter(todo =>
+    todo.title.toLowerCase().includes(search.toLowerCase()),
+  );
+
+  switch (sortBy) {
+    case Status.Active:
+      visibleTodos = visibleTodos.filter(todo => !todo.completed);
+      break;
+
+    case Status.Completed:
+      visibleTodos = visibleTodos.filter(todo => todo.completed);
+      break;
+
+    default:
+      break;
+  }
+
+  useEffect(() => {
+    if (titleInput.current) {
+      titleInput.current.focus();
+    }
+
+    getTodos()
+      .then(setTodos)
+      .catch(() => setErrorMessage('Unable to load todos'))
+      .finally(() => {
+        wait(3000).then(() => setErrorMessage(''));
```

Comment:
it's unnecessary action in every `finally`

better to create useEffect with timeout (and check - if errorMessage is not empty - set it empty)
One more variant is to create a function that sets some error **and** launch timeout where error will be cleared. And call this function in `catch`